lib/elixir/test/elixir/kernel/expansion_test.exs

```diff
@@ -957,157 +957,6 @@ defmodule Kernel.ExpansionTest do
     end
   end
 
-  describe "cond" do
-    test "expands each clause" do
-      before_expansion =
-        quote do
-          cond do
-            x = 1 -> x
-            true -> x
-          end
-        end
-
-      after_expansion =
-        quote do
-          cond do
-            x = 1 -> x
-            true -> x()
-          end
-        end
-
-      assert expand(before_expansion) == after_expansion
-    end
-
-    test "does not share lexical scope between clauses" do
-      before_expansion =
-        quote do
-          cond do
-            1 -> import List
-            2 -> flatten([1, 2, 3])
-          end
-        end
-
-      after_expansion =
-        quote do
-          cond do
-            1 -> :"Elixir.List"
-            2 -> flatten([1, 2, 3])
-          end
-        end
-
-      assert expand(before_expansion) == after_expansion
```

Comment:
This test is very important. It would change semantics if it fails.

Reply:
I believe that this cond is leaking:
```
cond do
  x = 1 -> x
  true -> x
end
```

Because it becomes:
```
case x = 1 do
  value when value != nil and value != false -> x
  _ -> x # this has the same context as the first x
end
```

<img width="1436" alt="screen shot 2018-05-23 at 00 01 34" src="https://user-images.githubusercontent.com/2848166/40393195-c2406a86-5e1e-11e8-809c-2dbc5873b230.png">

But I have the impression that we can just move the matches to the clauses

Reply:
Hrm... so maybe we can't rewrite a `cond` to a `case` after all? I see no way to address this. :(

Reply:
I don't see either, but the question that keeps hammering in my head is if this is a problem for macros in general?

If I understood correctly, it means that everytime you pass clauses to a macro, the context between clauses will be the same unless you find a way to separate the contexts in the generated code